spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/DenseTensor.scala

```diff
@@ -1717,24 +1720,64 @@ private[tensor] class DenseTensor[@specialized T: ClassTag](
     val indicesTensor = if (indices == null) Tensor[T]() else indices
     indicesTensor.resize(topKSize)
 
+    @inline
+    def compare(a: T, b: T): Boolean = if (increase) {
+      ev.isGreater(b, a)
+    } else {
+      ev.isGreater(a, b)
+    }
+
+    class LocalComparator(increase: Boolean) extends Comparator[(T, Int)] {
```

Comment:
why name it `Local` ?

Reply:
it's better not define a separate class, just define an anonymous class when declaring the TreeSet

Reply:
Because it's used only in this method.

Reply:
Good idea.